src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -527,7 +527,9 @@ private Completable stop(String errorMessage) {
             hubConnectionStateLock.unlock();
         }
 
-        return transport.stop();
+        Completable stop = transport.stop();
+        stop.subscribe();
```

Comment:
Can we also fix the LongPollingTransport so it returns the real Stop Completable from subsequent calls to stop() instead of returning Completable.complete()?

https://github.com/dotnet/aspnetcore/blob/d69a2253bb26b07290de3a7aa8d3bad54c610dfb/src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java#L172

If we did, this call to `stop.subscribe()` wouldn't be necessary, since we already do subscribe to stop(), it's sometimes just not the first call to stop().

https://github.com/dotnet/aspnetcore/blob/d69a2253bb26b07290de3a7aa8d3bad54c610dfb/src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java#L81

More importantly, this would prevents HubConnection.stop() from returning before the transport has actually stopped which is before the HubConnection can be safely restarted.

https://github.com/dotnet/aspnetcore/blob/d69a2253bb26b07290de3a7aa8d3bad54c610dfb/src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java#L530

Reply:
BTW I'm not saying we should remove this call to subscribe(). Do you have any idea why we currently call `.onErrorComplete().subscribe()` in the LongPollingTransport instead of just `.subscribe()` like we do here?